cabal-install/Distribution/Solver/Modular/Log.hs

```diff
@@ -72,17 +77,21 @@ logToProgress mbj l = let
     -- exhaustiveness and first conflict set.
     go :: Progress Message a b
        -> Progress Message a b
-       -> Progress String (Bool, Maybe (ConflictSet QPN)) b
+       -> Progress String (Exhaustive, ConflictSet QPN, ConflictMap) b
        -> Progress String String b
     go ms (Step _ ns) (Step x xs)           = Step x (go ms ns xs)
     go ms r           (Step x xs)           = Step x (go ms r  xs)
-    go ms _           (Fail (exh, Just cs)) = Fail $
-                                              "Could not resolve dependencies:\n" ++
-                                              unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
-                                              (if exh then "Dependency tree exhaustively searched.\n"
-                                                      else "Backjump limit reached (" ++ currlimit mbj ++
-                                                               "change with --max-backjumps or try to run with --reorder-goals).\n")
-                                                  where currlimit (Just n) = "currently " ++ show n ++ ", "
-                                                        currlimit Nothing  = ""
+    go ms _           (Fail (exh, cs, cm))  = Fail $
+        "Could not resolve dependencies:\n" ++
+        unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
```

Comment:
I like option (2).

> 1. either by choosing to print the first rather than the final conflict set also in the exhaustive case (what would this imply?),

I think that the first conflict set isn't very useful, because it comes from the first node where the solver backjumps, which is near the leaves of the tree.  It contains the same information as the first error.  However, the final conflict set is assigned to the root of the tree (in the exhaustive case), and it summarizes all conflicts in the tree.  It means that any solution to the dependency problem would need to assign a value that isn't available in the search tree to at least one of the variables in the conflict set.

The solver's conflict sets differ from the ones in [Modular lazy search for Constraint Satisfaction Problems] (http://dx.doi.org/10.1017/S0956796801004051) because they can also contain variables that are below the current node.  That difference makes the final conflict set more useful.

> 3 . or by printing the last error in the non-exhaustive case.

I'm not sure how this makes the behavior more consistent, or did you mean the exhaustive case?